OSS-24153 | Fixed type annotations in assignments not creating symbols

ofeki-neosec opened the pull request.

ofeki-neosec committed:
Fixed type annotations in assignments not creating symbols

varungandhi-src reviewed (commented):
I haven't had the time to look into this fully, so could you elaborate on:
- Is this meant as a full fix for the underlying bug?
- Is the 'TODO' meant for yourself or meant to be merged?

Additionally, could you add some tests covering different cases?

ofeki-neosec committed:
WIP

ofeki-neosec committed:
Fixed class fields

ofeki-neosec committed:
Fixed module attributes occurrences appearing twice in symbol ID

ofeki-neosec committed:
Fixed fields labels as builtins

ofeki-neosec committed:
Fixed constructors not getting occurrences

ofeki-neosec committed:
Fixed bug that caused missing class symbols

ofeki-neosec committed:
More type assumptions

ofeki-neosec committed:
Fixed missing ctor occurrences

ofeki-neosec committed:
Fixed bug with const references marked as something else

ofeki-neosec committed:
Small bugfixes

varungandhi-src committed:
fix: Avoid crash in the presence of decorators. (#184)

Changes the method lookup logic to use the same
logic as Pyright internals instead of our own ad-hoc
implementation of name resolution, which does not
match Python's method resolution order (MRO), which
is based purely on names, and not on types.

This gets rid of a code path where we were doing a `!`
operation, which triggered a crash when indexing
a prospect's codebase.

Fixes GRAPH-1278.

ofeki-neosec committed:
Added workaround unknown packages

ofeki-neosec committed:
Applied commit 8b60bbce1f2a4c7a517776cb395bbafb2e731e4f

ofeki-neosec committed:
Added index.py

ofeki-neosec committed:
Fixed cherry-pick

ofeki-neosec committed:
More fixes

ofeki-neosec committed:
Support non-utf-8 encoding

ofeki-neosec committed:
Fixed issue with assertions failing script
